### DISCUSSION TITLE
Documentation of Xerox keyboard.

### DISCUSSION THREAD

**Discussion by ecraven** [Interlisp/medley#193] [Show and tell]
The original Xerox 1108 keyboard:
![xerox-1108](https://user-images.githubusercontent.com/215510/108107221-3cf11980-708f-11eb-9de3-a33bf5a97e7b.png)
Based on https://www.pinterest.com/pin/30610472443101395/

---- EDIT -----
[the original below is *not* the correct keyboard, thanks @nbriggs for correcting me!]

I've drawn 
![xerox-1109](https://user-images.githubusercontent.com/215510/108070360-b9bace00-7064-11eb-8636-85291199e799.png)
based on https://web.archive.org/web/20201128140322/xahlee.info/kbd/xerox_1109_lisp_keyboard.html.
Is there a description anywhere on what all the keys are supposed to do?

**Comment by nbriggs** [comment]
That's NOT an 1109 keyboard.  This is an 1108/1109 keyboard:  [Dandelion keyboard](https://engblg.livingcomputers.org/wp-content/uploads/2019/01/image-4.png)

**Reply by ecraven** [reply]
ah, thank you very much. I'll draw that then ;) do you know whether there's any documentation on what the various keys do?

**Reply by nbriggs** [reply]
The keys do whatever they're programmed to do with their keyactions.  Have a look at the value of `\KEYNAMES`, each item represents a key, and `(KEYACTION 'keyname)` will tell you what's currently programmed for that key (e.g., `(KEYACTION 'RSHIFT)` will by default show `(2SHIFTDOWN . 2SHIFTUP)`, `(KEYACTION 'UNDERLINE)` is `((582 614 0 NOLOCKSHIFT) . IGNORE)`, and if TEdit has the keyboard focus it will interpret those codes per its readtable, and either add or remove an underline.  If the focus is in an EXEC window it will get the character codes directly because its readtable by default doesn't have anything connected to those codes.

**Comment by nbriggs** [comment]
After that came the Daybreak keyboard (the 1185/1186).   Prior to both was the Alto/Dolphin/Dorado keyboard, which didn't have the office keys (newer machines were repurposed 8010 (dandelion) and 6085 (daybreak) office systems)

**Reply by ecraven** [reply]
Is this https://upload.wikimedia.org/wikipedia/commons/a/aa/XeroxWorkstation.jpg a daybreak keyboard?

**Reply by nbriggs** [reply]
If you load the VIRTUALKEYBOARDS lispusers package and display the default keyboard you'll see the mapping that we used from the Dandelion keyboard functions to the Sun (type 3? 4?) keyboard.  The text editing keys were (are) set up to work with TEdit, so you can for example select a bit of text and use one of the property keys (ITALIC, etc.) to change things.

**Reply by nbriggs** [reply]
That's a Daybreak keyboard (1185/1186)

**Comment by masinter** [comment]
related to other keyboard issues
